core/src/main/java/org/geysermc/geyser/translator/protocol/java/level/JavaExplodeTranslator.java

```diff
@@ -49,14 +51,27 @@ public void translate(GeyserSession session, ClientboundExplodePacket packet) {
             ChunkUtils.updateBlock(session, BlockStateValues.JAVA_AIR_ID, pos);
         }
 
-        Vector3f pos = Vector3f.from(packet.getX(), packet.getY(), packet.getZ());
-        // Since bedrock does not play an explosion sound and particles sound, we have to manually do so
-        LevelEventPacket levelEventPacket = new LevelEventPacket();
-        levelEventPacket.setType(packet.getRadius() >= 2.0f ? LevelEventType.PARTICLE_HUGE_EXPLODE : LevelEventType.PARTICLE_EXPLOSION);
-        levelEventPacket.setData(0);
-        levelEventPacket.setPosition(pos);
+        LevelEventGenericPacket levelEventPacket = new LevelEventGenericPacket();
+        levelEventPacket.setEventId(2026/*LevelEventType.PARTICLE_BLOCK_EXPLOSION*/);
+        NbtMapBuilder builder = NbtMap.builder();
+        builder.putFloat("originX", packet.getX());
+        builder.putFloat("originY", packet.getY());
+        builder.putFloat("originZ", packet.getZ());
+        builder.putFloat("radius", packet.getRadius());
+        builder.putInt("size", packet.getExploded().size());
+        int i = 0;
+        for (Vector3i position : packet.getExploded()) {
+            Vector3i pos = Vector3i.from(packet.getX() + position.getX(), packet.getY() + position.getY(), packet.getZ() + position.getZ());
+            ChunkUtils.updateBlock(session, BlockStateValues.JAVA_AIR_ID, pos);
```

Comment:
This will send an UpdateBlockPacket and the LevelEventGenericPacket which I would assume also removes the blocks? If so, just replace this with a call to `session.getChunkCache().updateBlock`

Reply:
LevelEventGenericPacket is only responsible for the particles, which are more accurate then they would be with the LevelEventPacket, because it also displays the "block fizz"

Reply:
We haven't used the multi block update packet in the past because from what I understand there's zero benefit to using it. It's essentially a batched UpdateBlock*Synced*Packet (so two more fields to serialize).

Reply:
Ah sorry, didn't saw this must happened when I rebased